atlasdb-client/src/main/java/com/palantir/atlasdb/keyvalue/impl/DualWriteKeyValueService.java

```diff
@@ -104,6 +105,12 @@ public void putUnlessExists(TableReference tableRef, Map<Cell, byte[]> values)
     }
 
     @Override
+    public void checkAndSet(CheckAndSetRequest checkAndSetRequest) {
+        delegate1.checkAndSet(checkAndSetRequest);
+        delegate2.checkAndSet(checkAndSetRequest);
```

Comment:
I'd guess it's to ensure atomicity within cells - if you `putUnlessExists` to both `delegate1` and `delegate2`, then you might end up with exactly one of them succeeding. We probably want the same behaviour mirrored in the `checkAndSet` method.

This has slightly strange implications when a value for that cell exists in `delegate2`, but `checkAndSet` should only be mixed with `putUnlessExists` and never `put` (which we may want to enforce at some point due to #613)